File: experimental/workflow_mt/producer/rust/src/producer/implementation/emitters/finishconsumertest.rs
use super::{
    broadcast, events, hub, identification, producer::Control, protocol, unbound_pack, Context,
    EmitterError,
};
use clibri::server;
use uuid::Uuid;

pub async fn emit<E: server::Error, C: server::Control<E> + Send + Clone>(
    event: protocol::FinishConsumerTest,
    filter: hub::filter::Filter,
    context: &Context,
    control: &Control<E, C>,
) -> Result<(), EmitterError> {
    let mut broadcasting: Vec<(Vec<Uuid>, Vec<u8>)> = vec![];
    let mut broadcast_finishconsumertestbroadcast =
        events::finishconsumertest::emit::<E, C>(event, filter, context, control)
            .await
            .map_err(EmitterError::Emitting)?;
    broadcasting.push((
        broadcast_finishconsumertestbroadcast.0,
        unbound_pack(&0, &mut broadcast_finishconsumertestbroadcast.1)?,
    ));
    for msg in broadcasting.iter_mut() {
        broadcast::<E, C>(msg, control).await?;
    }
    Ok(())
}

File: tests/workflow/producer/typescript (copy)/src/producer/events/finishconsumertest.ts
import { Filter, Producer, Context, Protocol } from "../implementation/events";
import { Output } from "../implementation/events/finishconsumertest";

export function emit(
    event: Protocol.FinishConsumerTest,
    filter: Filter,
    context: Context,
    producer: Producer
): Promise<Output> {
    return Promise.reject(
        new Error(`Handler for event "FinishConsumerTest" isn't implemented`)
    );
}